src/Type.h

```diff
@@ -457,10 +457,19 @@ class IndexType : public Type {
 	const IntrusivePtr<TypeList>& GetIndices() const
 		{ return indices; }
 
-	[[deprecated("Remove in v4.1.  Use GetIndices().")]]
+	[[deprecated("Remove in v4.1. Use GetIndices().")]]
 	TypeList* Indices() const		{ return indices.get(); }
 
-	const std::vector<IntrusivePtr<Type>>& IndexTypes() const
+	[[deprecated("Remove in v4.1. Use GetIndexTypes().")]]
+	const type_list* IndexTypes() const
+		{
+		type_list* tl = new type_list();
```

Comment:
This method didn't used to return an allocated object for the caller to have to clean up, so I think what we'd actually need to do for backward compat is change `TypeList` to store a duplicate/mirrored `type_list` member.

Probably would want to do a sanity check of what impact the duplicate has on mem. usage too, but wouldn't guess Type size matters much there.